### PR TITLE
Upgrade Travis to use new container infrastructure

### PR DESCRIPTION
**Pull request by sarina**
See http://docs.travis-ci.com/user/migrating-from-legacy/?utm_source=legacy-notice&utm_medium=banner&utm_campaign=legacy-upgrade

### CODE REVIEW

**Comment by sarina** [comment]
@singingwolfboy or @jzoldak or @benpatterson maybe?

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
+sudo: false
 language: python
 branches:
     only:
         - master
 python: "2.7"
+cache: pip
```
Comment (benpatterson) [inline .travis.yml]: Unfo this won't work. See example [here](https://github.com/edx/ecommerce/blob/master/.travis.yml#L6).

And, a more concise example of a workaround is [here](https://github.com/semenko/https-everywhere/commit/1e17430a57a9bc8fe048757a4c03bf5dbe0479d1).

Comment (sarina) [inline .travis.yml]: Ugh, really? we've got a ton of repos to fix, then :cry:

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,12 @@
+sudo: false
 language: python
 branches:
     only:
         - master
 python: "2.7"
+cache:
+    directories:
+        - $HOME/.cache/pip
```
Comment (sarina) [inline .travis.yml:7]: @benpatterson this better?

**Comment by jzoldak** [comment]
@clintonb is making a [cookiecutter django template repo](https://github.com/clintonb/cookiecutter-django-edx/pull/1/files#diff-2c540bfcb102e51fbe600b5b48f35f0e).
Should we standardize on an order that we put these lines in?

**Comment by sarina** [comment]
I don't understand how the django template repo would relate to non Django based repos.

But yes, standardizing the order makes sense. However if it's going to be a lot of debate I'd like to just merge this and let that discussion happen elsewhere.

**Comment by clintonb** [comment]
@sarina @jzoldak I have no preference on order for `.travis.yml`. It's a file I rarely open.

**Comment by sarina** [comment]
I put the file in the same order as i18n-tools and other repos I use

**Comment by jzoldak** [comment]
:+1:

**Pull request merged**